rancher/configs.go

```diff
@@ -99,6 +99,27 @@ func createLaunchConfig(r *RancherService, name string, serviceConfig *config.Se
 	result.VolumeDriver = hostConfig.VolumeDriver
 	result.MilliCpuReservation = int64(serviceConfig.MilliCpuReservation)
 
+	if serviceConfig.RequestedHost != "" {
+		hosts, err := r.Client().Host.List(&client.ListOpts{
+			Filters: map[string]interface{}{
+				"name":         serviceConfig.RequestedHost,
+				"removed_null": nil,
+			},
+		})
+
+		if err != nil {
+			logrus.Errorf("Cannot list hosts: %v", err)
+			return result, err
```

Comment:
As a best practice, do not both log and return an error.
Only log the error if you are handing it and not bubbling it. Otherwise, just bubble it up.